pkg/kubectl/cmd/util/factory_builder.go

```diff
@@ -48,9 +49,21 @@ func NewBuilderFactory(clientAccessFactory ClientAccessFactory, objectMappingFac
 }
 
 func (f *ring2Factory) PrinterForCommand(cmd *cobra.Command) (printers.ResourcePrinter, bool, error) {
-	mapper, typer, err := f.objectMappingFactory.UnstructuredObject()
-	if err != nil {
-		return nil, false, err
+	var mapper meta.RESTMapper
+	var typer runtime.ObjectTyper
+	var err error
+
+	// if --local was set, avoid using UnstructuredObject as it makes
```

Comment:
This code block looks wrong.  If you want to know if the command is local pass an arg.  Then if you need to be local, then you should create local compatible mappers and typers.  The unstructured choice is more about decoding than anything else.